buffer/src/test/java/io/netty/buffer/RetainedSlicedByteBufTest.java

```diff
@@ -26,8 +25,7 @@
     protected ByteBuf newBuffer(int length, int maxCapacity) {
         Assume.assumeTrue(maxCapacity == Integer.MAX_VALUE);
         ByteBuf wrapped = Unpooled.wrappedBuffer(new byte[length * 2]);
-        ByteBuf buffer = wrapped.retainedSlice(length > 1 ?
-                        PlatformDependent.threadLocalRandom().nextInt(length - 1) + 1 : 0, length);
+        ByteBuf buffer = wrapped.retainedSlice(length > 1 ? 1 : 0, length);
```

Comment:
please note this is needed as otherwise you will sometimes get test errors as we may create a slice which is not large enough to write the requested bytes in. For example we may need 4 bytes and specify a length of 8 (which should be good enough). Now because of `threadLocalRandom().nextInt(...) + 1` we may end up with an offset that is >= 5 and so the write will fail.